### PR TITLE
Setup tooling to support React

### PR DESCRIPTION
**Pull request by mrgnr**
## Done

- Add esbuild for building react components
- Add `react` and `react-dom` packages
- Add TypeScript configuration
- Add a React component as a starting point for the Live Demo Box

## QA

- Open [demo](https://vanilla-framework-4312.demos.haus/docs/patterns/notification#live-demo-box)
- Verify that the Live Demo Box react component is rendered (see the screenshot below)

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [ ] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [ ] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical-web-and-design/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical-web-and-design/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [ ] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical-web-and-design/vanilla-framework/blob/main/templates/docs/whats-new.md).
- [ ] Documentation side navigation should be updated with the relevant labels.

## Issue

Fixes https://github.com/canonical-web-and-design/vanilla-squad/issues/1202

## Screenshots

react-live-demo-box

### CODE REVIEW

File: templates/_layouts/docs.html [hunk 1]
```diff
@@ -163,6 +163,7 @@
       window.Prism = window.Prism || {};
       Prism.manual = true;
     </script>
+    <script src="/static/js/dist/liveDemoBox.js" type="module"></script>
```
Comment (carkod) [inline templates/_layouts/docs.html:166]: Does this not work with `version_static` module?

File: build.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+const isDev = process && process.env && process.env.NODE_ENV === 'development';
+
+for (const [key, value] of Object.entries(entries)) {
+  const options = {
+    entryPoints: [value],
+    bundle: true,
+    minify: isDev ? false : true,
+    nodePaths: [path.resolve(__dirname, './templates/static/js/src')],
+    sourcemap: isDev ? false : true,
+    outfile: 'templates/static/js/dist/' + key + '.js',
```
Comment (bartaz) [inline build.js:17]: We currently build SCSS files to build/css directory (which is linked into static folder as `templates/static/build/css`) so for consistency maybe we can build JS into build/js, and it  should already be visible in static as `templates/static/build/js`). So we would avoid adding another "dist" folder and ignoring it.

File: templates/docs/patterns/notification.md [hunk 3]
```diff
@@ -10,6 +10,10 @@ context:
 
 Notifications are used to attract the user's attention. They offer four separate severity levels, and can also be modified for use in different contexts.
 
+## Live Demo Box
+
+<div id="react-live-demo-box"></div>
```
Comment (bartaz) [inline templates/docs/patterns/notification.md:15]: I know it's just a setup PR, so probably something to keep in mind later - we need to be able to have multiple live demo boxes on one page, so the script will have to run on multiple elements (like via "js-example" class name), not just one by id.

File: templates/_layouts/docs.html (hunk 1 above)
Comment (mrgnr) [reply templates/_layouts/docs.html]: I changed it to use `versioned_static`.

File: build.js (hunk 2 above)
Comment (mrgnr) [reply build.js]: I changed the path to `templates/static/build/js`.

File: templates/docs/patterns/notification.md (hunk 3 above)
Comment (mrgnr) [reply templates/docs/patterns/notification.md:15]: Good point, we'll keep this in mind :+1:

**Review by carkod — APPROVED** [review]
LGTM

**Comment by bartaz** [comment]
> @carkod The link check error happens when fetching the liveDemoBox.js bundle. I'm looking into it now. The fetch works for me locally using dotrun, but I think the bundle isn't copied to the right location in the Dockerfile.

Yes, seems Dockerfile may beed updating to copy the files correctly. Right now it fails to access liveDemoBox.js bundle on demo as well: 
Screenshot 2022-02-15 at 13 55 09

If it was just linting or tests failing I'd say it's ok to merge into feature branch, but it would be good to make sure demo is working, so Dockerfile may need a fix.

**Comment by mrgnr** [comment]
@carkod @bartaz It's working now -- I'd forgotten to update the path to the build in the Dockerfile.

**Review by bartaz — APPROVED** [review]
LGTM, thanks!

**Pull request merged**